pkg/idp/provider/azure/azure.go

```diff
@@ -396,28 +397,64 @@ func (a *Azure) GetCredentialsForAuthenticatedApp(config provider.AppConfig) (ma
 }
 
 func (a *Azure) CleanCredentialsForAuthenticatedApp(config provider.AppConfig) error {
-	for _, name := range []string{config.Name, DexOperatorName} {
-		app, err := a.GetApp(name)
-		if err != nil {
-			if !IsNotFound(err) {
-				return microerror.Mask(err)
-			}
-			continue
+	app, err := a.GetApp(config.Name)
+	if err != nil {
+		if !IsNotFound(err) {
+			return microerror.Mask(err)
 		}
-		id := app.GetId()
-		if id == nil {
-			return microerror.Maskf(notFoundError, "Could not find ID of app %s.", name)
+		return nil
+	}
+	id := app.GetId()
+	if id == nil {
+		return microerror.Maskf(notFoundError, "Could not find ID of app %s.", config.Name)
+	}
+	for _, c := range app.GetPasswordCredentials() {
+		if credentialName := c.GetDisplayName(); credentialName != nil {
+			if *credentialName == config.Name && secretChanged(c, a.clientSecret) {
+				if err = a.DeleteSecret(context.Background(), c.GetKeyId(), *id); err != nil {
+					return microerror.Mask(err)
+				}
+				a.Log.Info(fmt.Sprintf("Removed secret %v of %s app %s for %s in microsoft ad tenant %s", c.GetKeyId(), a.Type, config.Name, a.Owner, a.TenantID))
+			}
 		}
-		for _, c := range app.GetPasswordCredentials() {
-			if credentialName := c.GetDisplayName(); credentialName != nil {
-				if *credentialName == config.Name && secretChanged(c, a.clientSecret) {
-					if err = a.DeleteSecret(context.Background(), c.GetKeyId(), *id); err != nil {
-						return microerror.Mask(err)
-					}
-					a.Log.Info(fmt.Sprintf("Removed secret %v of %s app %s for %s in microsoft ad tenant %s", c.GetKeyId(), a.Type, config.Name, a.Owner, a.TenantID))
+	}
+	return nil
+}
+
+func (a *Azure) DeleteAuthenticatedApp(config provider.AppConfig) error {
```

Comment:
This function deletes all dex-apps for a given installation first and then the dex-operator itself.
We simply query by app name and installation name here. I would have liked to use owner app (delete all apps owned by app x) but unfortunately the ids used to reference the owner can not be used to query for the app itself.